container/config.py

```diff
@@ -44,18 +44,28 @@ class AnsibleContainerConfig(Mapping):
     base_path = None
 
     @container.host_only
-    def __init__(self, base_path, var_file=None, engine_name=None):
+    def __init__(self, base_path, var_file=None, engine_name=None, project_name=None):
         self.base_path = base_path
         self.var_file = var_file
         self.engine_name = engine_name
         self.config_path = path.join(self.base_path, 'container.yml')
+        self.project_name = project_name
         self.set_env('prod')
 
     @property
     def deployment_path(self):
         dep_path = self.get('settings', yaml.compat.ordereddict()).get('deployment_output_path',
                             path.join(self.base_path, 'ansible-deployment/'))
-        return path.normpath(path.abspath(path.expanduser(dep_path)))
+        return path.normpath(path.abspath(path.expanduser(path.expandvars(dep_path))))
+
+    @property
+    def image_namespace(self):
+        # When pushing images or deploying, we need to know the registry namespace
+        namespace = self.project_name
+        if self.engine_name in ('k8s', 'openshift'):
```

Comment:
I need to know what `config` thinks the namespace is, so seemed logical to add a single method to the config object. There is prior art within `config` referencing the engine_name value, so not sure why it's suddenly an issue here.